Medical Uniform and Vend Update, adds Field Doctor Scrubs & Uniform

Puckaboo2 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->

# About the pull request

Everything has been implemented; checks mean they have been tested. Draft until #11357 merges so I can add the new pouches to Surgeon, Pharmphys, and Field Doctor.

**PART ONE: Doctor Job variant differentiation**

- [x] Surgeon, Pharmacist, and Field Doctor have unique recommendations in the ColMarTech Medical Equipment Rack for their loadout, just as Doctors, Nurses, and Researchers do.
- [x] Gives Field Doctor an Essentials set that includes almost every item in the Field Doc's locker.
- [x] Surgeon and Pharmaceutical Physician have unique radio names, icon tags, and icons on the minimap. You will never confuse your fellow doctors, again!
- [x] Adds a surgical mask for everybody's essentials.

**PART TWO: Lab Coat Update**

- [x] Buffs `/obj/item/clothing/suit/storage/labcoat`; they all have some sort of melee, bullet, laser, stamina, and bomb armor.
- [x] Modifies descriptions.
- [x] Updates lab coat shading.
- [x] Removes Pharmaceutical Physician lab coat, renames it to Chemistry lab coat, and removes it from chemistry wardrobe. I know, I will miss it, too.
- [x] Updates snow suit sprites.

**PART THREE: Medical uniform update**
- [x] Reorganizes contents in the ColMarTech Medical Equipment Rack for each medical role based on item type and loadout item recommendation.
- [x] Most Medical Uniforms have been re-shaded and recolored.
- [x] Adds Field Doctor USCM uniform and Field Doctor scrubs.
- [x] Gives a unique color palette and bio resist to the medical apron.
- [x] **Nukes a handful of pouches  (Medbay Only)**
- [x] Medical scrubs and caps no longer have job names attached to them (CMO gets to keep theirs tho.) Unfortunately, that means PharmPhys' scrubs and cap had to go. I'm sorry. I'll miss them, too. You get to keep White, though, prommy.

**PART FOUR: Research Vend Update**
- [x] Lumps sames with sames, uniforms with uniforms, accessories with accessories, all in one category.
- [x] Adds a longer science coat.
- [x] Reorganizes categories.
- [x] Capitalizes vending items.
- [x] Adds webbings and pouches to research vending options.
- [x] Adds a webbing vest to Research Essentials 

# Explain why it's good for the game

**PART ONE: Doctor Job variant differentiation**

1. Surgeon, Field Doctor, and Pharmacist are not simply flavors of a doctor; while they have the same skillset, their interests, gear, and priorities differ, and their loadouts should accommodate those differences. For example, a Pharmacist will be recommended a Chemistry pouch, but Doctor, Surgeon, Field Doc, and Nurse will not.
2. Replacement field docs should have the comforts of the previous doctor, so I included the locker items as part of a new essentials bundle called 'Medical and Field Essentials.'
3. We will never mistake a surgeon for a pharmacist ever again.
4. Eh... why not? Every doctor should have one of them.

**PART TWO: Lab Coat Update**
1. Having a coat is better than nothing, so you should have some resistance to basic types (parent has verylow in most of them). Plus, CMO's cost is described as 'sturdy,' so it deserves a little more defense. Finally, W-Y's lab coat is considered the best of the best lab coats out there and should not be of lesser quality and provide fewer defenses than a standard snowsuit, which it currently does.
2. Lab coats and scrubs now say they protect against chemical mishaps and biological hazards in the description.
3. I re-shaded them before and I felt like doing it again. Lab coats are now Medical White instead of Surgery Off-White.
4. Maintainers have decreed the Chemist labcoats should not be added for PharmPhys. If I had known it would have been an issue, I would never have assigned them to chemist wardrobes. They are nuked from PharmPhys loadouts and chemist wardrobes. I am sorry.
5. The side sprites looked kind of... old... And it had like 20 colors in its palette when it only needed around 8. And those side sprites--the first word I thought of was 'uggo' and I felt compelled to fix them.

**PART THREE: Medical uniform update**
1. Some items felt improperly arranged, visually and practically speaking. For ease of use, hats are now situated below eyewear, and, to assist newbies, items are arranged by Mandatory -> Recommended -> Normal from top to bottom.
2. Surgical caps and scrubs had a metallic appearance to them; this originated from the original bay sprites for surgical scrubs and caps, which I replicated during the resprite. Now, they're shaded to appear like soft, light fabric. I also restored the original scrub colors that I tweaked in the PharmPhys update.
3. Camo scrubs or robust field doc uniform? That is for the field doctor to decide. It feels odd for field docs to go groundside without an official camo uniform to utilize.
4. Once again... Why not? In all actuality, though, there was not a single difference between the medical apron and the chef's apron other than name. . It also has a little bit of bio resist, as a treat.
5. Most of these pouches are for the marine medics who do not have enough training to use the advanced pouches doctors know and love on every day, so I thought it would be best to get rid of them for medbay personnel only.
6. Having colors assigned to job roles was an oversight that has now been addressed to me, so pharmphys uniforms were nuked. I'm sad, too. If I had known color-codes for medical jobs were not supposed to be a thing, I would not have gone through with it.

**PART FOUR: Research Vend Update**
1. There should not be 3 categories for undersuits; a category implies you can take one suit from each, which is simply not the case. Anything 'undersuit' and 'accessory' related is lumped into one category.
2. Why not add a longer brown lab coat? It looks cool. 
3. Reorganizing the categories retains categorical consistency with the other medical staff's vending options.
4. There were inconsistencies with word capitalization, so I capitalized all words to retain consistency.
5. ... Why remove webbing accessories from researchers? Why? Seriously... Why? Because the creator of the PR said 'I don't need them so the rest of the community does not need them'? Miss me with that. Here you go, fellas, you can have your webbings back. 
6. If you chose a tie, you can still have a webbing, and vice-versa.

# Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->

Surgical mask is now in essentials 
<img width="197" height="206" alt="image" src="https://github.com/user-attachments/assets/0f8287e2-13d7-40af-9a1e-fce2d3a47434" />

Field Doctor's Essentials 
<img width="231" height="326" alt="image" src="https://github.com/user-attachments/assets/e83e4f2f-67a8-41ca-8ab2-dd2fae3d2688" />

Field Doctor's Essentials (I fixed the surgical mask thing)
<img width="413" height="737" alt="image" src="https://github.com/user-attachments/assets/2fed5502-c2ad-425f-be50-39188fcd047f" />

Snow Suit Sprite
<img width="405" height="151" alt="image" src="https://github.com/user-attachments/assets/1ace8b28-c56f-4cc4-9dfa-da6cd4192a31" />

Job Role prioritizations
<img width="417" height="783" alt="image" src="https://github.com/user-attachments/assets/125f9b61-ce0a-4c50-b999-26cf81c86590" />

Field Doctor, for example.
<img width="436" height="620" alt="image" src="https://github.com/user-attachments/assets/258743bb-ced2-4dd2-b40b-a14ae9805bb1" />

Pouches and webbings have prioritizations
<img width="437" height="791" alt="image" src="https://github.com/user-attachments/assets/6d8de742-9ea4-489c-96d2-c62b1f6295cb" />

<!-- !! If you are modifying sprites, you **must** include one or more in-game screenshots or videos of the new sprites. !! -->

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>


# Changelog

:cl: Puckaboo2
add: Each job role in medical has its own loadout with recommendations.
add: Adds a medium-length brown research coat.
add: Adds Field Doctor's locker contents into the vended essentials or vend loadout in case the replacement finds nothing in their locker.
add: Nurses and CMOs can have the same pouches as doctors.
add: Adds a sterile mask as part of medical staff's essentials.
add: Researchers can find white lab coats in their vending options.
add: Nurses can find lab coats in their vending options.
add: Adds a field doctor USCM uniform and surgical scrubs that camo like marine uniforms. Wear the scrubs if you want more bio protection, wear the uniform if you want to be more robust.
balance: Due to the dangers medical personnel are expected to encounter, most lab coats are now built with basic protection against most types of damage, while Chief's and W-Y's lab coats are even more robust. This ain't Baystation, boys.
balance: Added the same amount of bio and rad resist to CMO and RD's regular uniforms.
balance: Most often, scrubs vs. medical uniforms have the same amount of bio resist.
code: Reorganized the medical equipment rack's items based on recommendation.
code: Tweaks scrub descriptions to say they protect against radiation.... Because they do.
del: Pharmaceutical Physician scrubs and caps have been removed from the game. The scrubs were repurposed as the Chemistry uniform.
del: Pharmaceutical Physician lab coats have been removed from PharmPhys loadouts and the chemical wardrobe. They were repurposed as chemistry lab coats.
del: Nuked a handful of pouches from Medical's ColMarTech machine. Medbay only. Don't worry, corpsmen, I didn't touch your stuff.
del: Replaced grey scrubs with black scrubs in the white medical wardrobe.
imageadd: Adds a sterile2 mask onmob spite. Dunno what sterile2 is for, but it's there now.
imagetweak: Most of the medical clothing has been re-shaded.
imagetweak: Updated snowsuit sprites to be more soft, plush, and cozy. Try one on!
maptweak: Removed the chemistry wardrobe in the chem bay. It was getting stuffy in there.
qol: Hats are below eyewear and above uniforms in medical equipment vendors.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! -->

Puckaboo2 committed:
reorganizing

Puckaboo2 committed:
icon

Puckaboo2 committed:
Implements labcoats

implements labcoats

More fixes

More fixes

Puckaboo2 committed:
Buffs some lab coats

implements labcoats

More fixes

More fixes

Puckaboo2 committed:
Buffs some lab coats

implements labcoats

More fixes

More fixes

Puckaboo2 committed:
Forgot to add field doctor essential set

Puckaboo2 committed:
more changes

Puckaboo2 committed:
More additions

Puckaboo2 committed:
Merge branch 'master' into researchcoats

Puckaboo2 committed:
More fixes

Puckaboo2 committed:
Fixes more sprites

Puckaboo2 committed:
Fixes more sprites

Puckaboo2 committed:
Update medical.dm

Puckaboo2 committed:
Trying to make the linter happy

Puckaboo2 committed:
Update medical.dm

Puckaboo2 committed:
Update medical.dm

Puckaboo2 committed:
Update medical.dm

Puckaboo2 committed:
Uniform update

Huge medical sprite update

Uniform update

Fixes coats again

Drulikar commented on code/game/machinery/vending/vendor_types/crew/medical.dm:
```diff
@@ -260,50 +522,80 @@ GLOBAL_LIST_INIT(cm_vending_clothing_researcher, list(
+		/obj/item/clothing/shoes/marine,
+		/obj/item/storage/box/mre,
+		/obj/item/device/radio/marine,
+//		/obj/item/clothing/head/helmet/marine/medic,  this caused '/obj/item/device/helmet_visor/medical/advanced was unable to be GC'd'. Don't remove it. Somebody figure out why it happened - Puckaboo2
```
Remove this comment and apply the patch that fixes your issue here. I don't care ultimately where you put the helmet, but it still should be fixed. 
[patch.diff.txt](https://github.com/user-attachments/files/24224555/patch.diff.txt)

Puckaboo2 committed:
Fixes visors unable to be GC'd

Puckaboo2 committed:
replaces lab coat

Puckaboo2 committed:
Create 11324_repath_science_labcoat.txt

Puckaboo2 committed:
Update labcoat.dm

Puckaboo2 committed:
medical apronsss

Puckaboo2 committed:
codes in medical aprons

Puckaboo2 committed:
updates snow suit sprites

Puckaboo2 committed:
Update vests_aprons.dmi

Puckaboo2 committed:
Small spell check, not worth changelogging.

Puckaboo2 committed:
Update medical.dm

Puckaboo2 committed:
Adds field doctor uniform

Adds field doctor uniform

Puckaboo2 committed:
Fixed an oopsie with the desert uniforms.

Puckaboo2 committed:
Fixed an oopsie with the desert uniforms.

Puckaboo2 committed:
Adds med roles to global list init

Puckaboo2 committed:
Fixes pharmphys shading

Puckaboo2 committed:
adds more things to belts

Puckaboo2 committed:
Update medical.dm

Puckaboo2 committed:
More fixes before testing

More fixes before testing

Puckaboo2 committed:
fixes shading on field doc uniforms

Puckaboo2 committed:
fixes shading on field doc uniforms

Puckaboo2 committed:
Update coats_robes.dmi

Puckaboo2 committed:
Some more sprite updates

Puckaboo2 committed:
Update medsci.dm

Puckaboo2 committed:
adds more bio resist to uniforms

Puckaboo2 committed:
Merge branch 'researchcoats' of https://github.com/Puckaboo2/cmss13 into researchcoats

Puckaboo2 committed:
More sprite edits

Puckaboo2 committed:
Revert "adds more things to belts"

This reverts commit a11e68d1a64d49ce10aa7c7f2545d725337b9309.

Puckaboo2 committed:
Update coats_robes.dmi

Puckaboo2 committed:
Merge remote-tracking branch 'upstream/master' into researchcoats

Puckaboo2 committed:
Reorganization

Puckaboo2 committed:
Reorganization and renaming

Puckaboo2 committed:
Some more changes

Puckaboo2 committed:
Removes uniforms we no longer use

Puckaboo2 committed:
Revert "Removes uniforms we no longer use"

This reverts commit 649d4616e210655ff13bf739727102892c341e01.

Puckaboo2 committed:
Update medical.dmi

Puckaboo2 committed:
Establishes other roles in minimap and HUD

Puckaboo2 committed:
huge batch of changes

Puckaboo2 committed:
huge batch of changes

Puckaboo2 committed:
huge batch of changes

Puckaboo2 committed:
Makes linter happy

Puckaboo2 committed:
Update BigRed.dmm

Puckaboo2 committed:
Some more changes

Puckaboo2 committed:
Some more changes

Puckaboo2 committed:
Update senior_officers.dm

Puckaboo2 committed:
Update labcoat.dm

Puckaboo2 committed:
Fixup maps in TGM format

35c0ab854d9e: maps/map_files/BigRed/BigRed.dmm

Automatically commited by: tools\mapmerge2\fixup.py

Puckaboo2 committed:
Update wardrobe.dm

Puckaboo2 committed:
Update medical.dm

Puckaboo2 committed:
Some more changes

Puckaboo2 committed:
adds oldschool uniforms and coats back

Puckaboo2 committed:
don't need these, they're in another PR

Puckaboo2 committed:
Update medical.dm

Puckaboo2 committed:
Update USS_Almayer.dmm

Puckaboo2 committed:
Merge branch 'master' into researchcoats

Puckaboo2 committed:
Unalphabetizes stuff

Puckaboo2 committed:
Reverses some things, changes others

Puckaboo2 committed:
Update medical.dm

Puckaboo2 committed:
Merge remote-tracking branch 'upstream/master' into researchcoats

Puckaboo2 committed:
Update vests_aprons.dmi

Puckaboo2 committed:
Merge remote-tracking branch 'upstream/master' into researchcoats

cm13-github commented:
Conflicts have been resolved. A maintainer will review the pull request shortly.

Puckaboo2 committed:
Map fix

Puckaboo2 committed:
More edits

Puckaboo2 committed:
Merge remote-tracking branch 'upstream/master' into researchcoats

Puckaboo2 committed:
Update colonialmarines.dme

Puckaboo2 committed:
pls fix

Puckaboo2 committed:
Fixes some lists

Puckaboo2 committed:
Some more fixes

Puckaboo2 committed:
Forgot to add the green collar

Nanu308 reviewed (dismissed):
mapping approved

Puckaboo2 committed:
Streamlines pouches for doctors

Puckaboo2 committed:
Merge branch 'master' into researchcoats

Puckaboo2 committed:
Merge branch 'master' into researchcoats

Puckaboo2 committed:
Update vests_aprons.dmi

Puckaboo2 committed:
Merge branch 'researchcoats' of https://github.com/Puckaboo2/cmss13 into researchcoats

Puckaboo2 committed:
Merge branch 'master' into researchcoats

Puckaboo2 committed:
Fixed some stuff

cm13-github commented:
This pull request has conflicts, please resolve those before we can evaluate the pull request.

Puckaboo2 committed:
Merge remote-tracking branch 'upstream/master' into researchcoats

cm13-github commented:
Conflicts have been resolved. A maintainer will review the pull request shortly.

Puckaboo2 committed:
Why was this changed?

Puckaboo2 committed:
Fixes missing things

Puckaboo2 committed:
Woops

Puckaboo2 committed:
Merge remote-tracking branch 'upstream/master' into researchcoats

Puckaboo2 committed:
Update jobs.dm

cm13-github commented:
This pull request has conflicts, please resolve those before we can evaluate the pull request.

Puckaboo2 committed:
Merge remote-tracking branch 'upstream/master' into researchcoats

cm13-github commented:
Conflicts have been resolved. A maintainer will review the pull request shortly.
